internal/reliable_topic.go

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"time"
+
+	"sync/atomic"
+
+	"sync"
+
+	"strings"
+
+	"strconv"
+
+	"log"
+
+	"github.com/hazelcast/hazelcast-go-client/config"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	"github.com/hazelcast/hazelcast-go-client/internal/iputil"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto"
+	"github.com/hazelcast/hazelcast-go-client/internal/proto/bufutil"
+	"github.com/hazelcast/hazelcast-go-client/internal/reliabletopic"
+	"github.com/hazelcast/hazelcast-go-client/internal/serialization"
+)
+
+const (
+	topicRBPrefix  = "_hz_rb_"
+	maxBackoff     = 2000 * time.Millisecond
+	initialBackoff = 100 * time.Millisecond
+)
+
+type ReliableTopicProxy struct {
+	*proxy
+	ringBuffer           core.Ringbuffer
+	serializationService *serialization.Service
+	topicOverLoadPolicy  core.TopicOverloadPolicy
+	config               *config.ReliableTopicConfig
+	msgProcessorsMu      *sync.Mutex
+	msgProcessors        map[string]*messageProcessor
+}
+
+func newReliableTopicProxy(client *HazelcastClient, serviceName string, name string) (*ReliableTopicProxy, error) {
+	proxy := &ReliableTopicProxy{
+		msgProcessors:   make(map[string]*messageProcessor),
+		msgProcessorsMu: new(sync.Mutex),
+		proxy: &proxy{
+			client:      client,
+			serviceName: serviceName,
+			name:        name,
+		},
+	}
+	proxy.serializationService = client.SerializationService
+	// TODO:: get reliable topic config with ringbuffer id
```

Comment:
This seems important for this feature. I think, It should be handled with this pr.

Reply:
I think I forgot to remove the TODO, because we already get the reliable config with its name which is its id.